wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -1011,24 +1011,14 @@ private void createLogsDir() {
     private void cleanup() {
         eventService.unsubscribe(machineCleaner);
 
-        boolean interrupted = false;
         executor.shutdown();
-        try {
-            if (!executor.awaitTermination(10, TimeUnit.SECONDS)) {
-                executor.shutdownNow();
-                if (!executor.awaitTermination(10, TimeUnit.SECONDS)) {
-                    LOG.warn("Unable terminate main pool");
-                }
-            }
-        } catch (InterruptedException e) {
-            interrupted = true;
-            executor.shutdownNow();
-        }
 
         try {
             for (MachineImpl machine : machineRegistry.getMachines()) {
                 try {
                     destroy(machine.getId(), false);
+                } catch (NotFoundException ignore) {
+                    // if machine was destroyed before, it is ok
```

Comment:
I'd say that it's better to use comment like `it is ok, machine has been already destroyed`